velox/functions/lib/aggregates/MinMaxByAggregatesBase.h

```diff
@@ -0,0 +1,650 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/exec/Aggregate.h"
+#include "velox/exec/ContainerRowSerde.h"
+#include "velox/functions/lib/aggregates/SingleValueAccumulator.h"
+#include "velox/functions/prestosql/aggregates/AggregateNames.h"
```

Comment:
There is a circular dependency here: lib depends on prestosql, which depends on lib. Please, resolve.

Reply:
Done. Would you help review again?